src/index.js

```diff
@@ -0,0 +1,467 @@
+    });
+
+    // If there are no inputs, set state where form is ready to trigger
+    // change event. New inputs might be added later
+    if (!this.inputs.length) {
+      this.setState({
+        canChange: true,
+      });
+    }
+  }
+
+  render() {
+    const {
+      getErrorMessage,
+      getErrorMessages,
+      getValue,
+      hasValue,
+      isFormDisabled,
+      isFormSubmitted,
+      isPristine,
+      isRequired,
+      isValid,
+      isValidValue,
+      mapping,
+      onChange,
+      // onError,
+      onInvalidSubmit,
+      onInvalid,
+      onReset,
+      onSubmit,
+      onValid,
+      onValidSubmit,
+      preventExternalInvalidation,
+      // reset,
+      resetValue,
+      setValidations,
+      setValue,
+      showError,
+      showRequired,
+      validationErrors,
+      ...nonFormsyProps
+    } = this.props;
+
+    return React.createElement(
+      'form',
+      {
+        onReset: this.resetInternal,
+        onSubmit: this.submit,
+        ...nonFormsyProps,
+      },
+      this.props.children,
+    );
+  }
+}
+
+Formsy.displayName = 'Formsy';
+
+Formsy.defaultProps = {
+  children: null,
+  disabled: false,
+  getErrorMessage: () => {},
+  getErrorMessages: () => {},
+  getValue: () => {},
+  hasValue: () => {},
+  isFormDisabled: () => {},
+  isFormSubmitted: () => {},
+  isPristine: () => {},
+  isRequired: () => {},
+  isValid: () => {},
+  isValidValue: () => {},
+  mapping: null,
+  onChange: () => {},
+  onError: () => {},
+  onInvalid: () => {},
+  onInvalidSubmit: () => {},
+  onReset: () => {},
+  onSubmit: () => {},
+  onValid: () => {},
+  onValidSubmit: () => {},
+  preventExternalInvalidation: false,
+  resetValue: () => {},
+  setValidations: () => {},
+  setValue: () => {},
+  showError: () => {},
+  showRequired: () => {},
+  validationErrors: null,
+};
+
+Formsy.propTypes = {
+  children: PropTypes.node,
+  disabled: PropTypes.bool,
+  getErrorMessage: PropTypes.func,
+  getErrorMessages: PropTypes.func,
+  getValue: PropTypes.func,
+  hasValue: PropTypes.func,
+  isFormDisabled: PropTypes.func,
+  isFormSubmitted: PropTypes.func,
+  isPristine: PropTypes.func,
+  isRequired: PropTypes.func,
+  isValid: PropTypes.func,
+  isValidValue: PropTypes.func,
+  mapping: PropTypes.object, // eslint-disable-line
+  preventExternalInvalidation: PropTypes.bool,
+  onChange: PropTypes.func,
+  onInvalid: PropTypes.func,
+  onInvalidSubmit: PropTypes.func,
+  onReset: PropTypes.func,
+  onSubmit: PropTypes.func,
+  onValid: PropTypes.func,
+  onValidSubmit: PropTypes.func,
+  resetValue: PropTypes.func,
+  setValidations: PropTypes.func,
+  setValue: PropTypes.func,
+  showError: PropTypes.func,
+  showRequired: PropTypes.func,
+  validationErrors: PropTypes.object, // eslint-disable-line
+};
+
+Formsy.childContextTypes = {
+  formsy: PropTypes.object,
+};
+
+const addValidationRule = (name, func) => {
+  validationRules[name] = func;
```

Comment:
This will overwrite any existing validation rule with the same name.  This can lead to unexpected behaviour, especially when working on large projects or projects with multiple developers.  The following snippet will warn the user if they attempt to add a validation rule with a name that already exists.  There's also an override parameter to allow the user to forcibly overwrite existing validation rules.

```js
Formsy.addValidationRule = function (name, func, force = false) {
  if (!force && Object.keys(validationRules).indexOf(name) !== -1) {
    console.error(`A Validation Rule with the name '${name}' already exists`);
    return;
  }
  validationRules[name] = func;
};
```